### PR TITLE
feat: add alert when missing attachment #1220

### PR DESCRIPTION
**Pull request by mgrigoriev8109**
# 🔗 Issue
https://github.com/rubyforgood/homeward-tails/issues/1221

# ✍️ Description
Adds an alert when a staff user attempts to upload 0 files or images

# 📷 Screenshots/Demos
`bin/dev`, log in as Staff, head to http://localhost:3000/alta/staff/pets/2?active_tab=photos 

Click Attach without any photo uploaded, and see the new alert.
Screenshot 2024-12-17 at 3 04 09 PM

Add an image and check that the success alert pops up as it was intended.
Screenshot 2024-12-17 at 3 04 35 PM

Do the same for Files
Screenshot 2024-12-17 at 3 05 06 PM
Screenshot 2024-12-17 at 3 05 24 PM

### CODE REVIEW

**Comment by mgrigoriev8109** [comment]
Need to tinker with this a bit more, should've ran the tests locally - looks like having my method as a before_action for those actions is breaking some stuff.

File: app/controllers/concerns/attachment_manageable.rb [hunk 1]
```diff
@@ -5,4 +5,15 @@ def allow_only_one_attachment
     flash.now[:alert] = t(".attachment_manageable.upload_limit_exceeded")
     render turbo_stream: turbo_stream.replace("flash", partial: "layouts/shared/flash_messages")
   end
+  
+  def alert_for_no_images
+    if params[:action] === 'attach_images' and params[:pet][:images].length > 1
```
Comment (kasugaijin) [inline app/controllers/concerns/attachment_manageable.rb:10]: `==` in Ruby if you want equality
https://dev.to/baweaver/understanding-ruby-triple-equals-2p9c `===` in Ruby means something else

**Review by kasugaijin — COMMENTED** [review]
Looks good! Just a couple small comments.

File: app/controllers/concerns/attachment_manageable.rb
Comment (kasugaijin) [inline app/controllers/concerns/attachment_manageable.rb:1]: For the tests that are failing, you might want to throw a debugger in the tests to see what the params look like in the test context. It looks like what you are doing here to check if there is an attachment works in the browser, but not in the tests.

**Comment by mgrigoriev8109** [comment]
@kasugaijin this is ready for a re-review. The `:images` and `:files` params behaved differently, just like you predicted. 

In a live environment for some reason they're populated with an empty string, so they become an array of `["", {image-object}, {image-object}]`. 

I figured instead of adding empty strings to the test environment it'd be cleaner to just remove those empty strings in the method. Let me know if this works, or if I should implement it differently.

File: app/controllers/concerns/attachment_manageable.rb [hunk 2]
```diff
@@ -2,7 +2,24 @@ module AttachmentManageable
   def allow_only_one_attachment
     return unless params[:files].is_a?(Array)
 
-    flash.now[:alert] = t(".attachment_manageable.upload_limit_exceeded")
+    flash.now[:alert] = t("attachment_manageable.upload_limit_exceeded")
     render turbo_stream: turbo_stream.replace("flash", partial: "layouts/shared/flash_messages")
   end
+
+  def show_alert_if_attachment_missing
+    if !params[:pet][:images].nil?
```
Comment (kasugaijin) [inline app/controllers/concerns/attachment_manageable.rb:10]: I think it would be more idiomatic to use
`if params[:pet][:images].present?` rather than a double negative

File: app/controllers/concerns/attachment_manageable.rb [hunk 3]
```diff
@@ -2,7 +2,24 @@ module AttachmentManageable
...
+      no_empty_images = params[:pet][:images].reject { |image| image == "" }
```
Comment (kasugaijin) [inline app/controllers/concerns/attachment_manageable.rb:11]: `no_empty_images` could just be renamed `images`, same for `files`.

File: app/controllers/concerns/attachment_manageable.rb [hunk 4]
```diff
@@ -2,7 +2,24 @@ module AttachmentManageable
...
+    elsif !params[:pet][:files].nil?
+      no_empty_files = params[:pet][:files].reject { |image| image == "" }
```
Comment (kasugaijin) [inline app/controllers/concerns/attachment_manageable.rb:13]: I'd also add a comment on why we are rejecting the ""

File: app/controllers/concerns/attachment_manageable.rb (hunk 2 above)
Comment (mgrigoriev8109) [reply app/controllers/concerns/attachment_manageable.rb]: Did this in [refactor: use present? instead of nil?](https://github.com/rubyforgood/homeward-tails/pull/1273/commits/2454e2d0883c9115aa2f2132d03d65f9dd1ba5a8)

File: app/controllers/concerns/attachment_manageable.rb (hunk 3 above)
Comment (mgrigoriev8109) [reply app/controllers/concerns/attachment_manageable.rb]: Did this in [refactor: change no_empty_images to images and files](https://github.com/rubyforgood/homeward-tails/pull/1273/commits/22d84b32d7593e16f3339229da03991124a44a7f)

File: app/controllers/concerns/attachment_manageable.rb (hunk 4 above)
Comment (mgrigoriev8109) [reply app/controllers/concerns/attachment_manageable.rb]: Did this in [chore: add comment on why we remove empty string from files / images …](https://github.com/rubyforgood/homeward-tails/pull/1273/commits/3e3bdb83b7e572b9a2baef81bff2f67f11da91fb)

**Review by kasugaijin — APPROVED** [review]
@mgrigoriev8109 thank you!
